zbus/tests/e2e.rs

```diff
@@ -277,6 +277,21 @@ impl MyIface {
         header.is_some()
     }
 
+    #[instrument]
+    #[zbus(property)]
+    fn set_test_header_prop(
+        &self,
+        value: bool,
+        #[zbus(header)] header: Option<Header<'_>>,
+        #[zbus(connection)] connection: &Connection,
+        #[zbus(object_server)] object_server: &ObjectServer,
+        #[zbus(signal_emitter)] emitter: SignalEmitter<'_>,
+    ) {
+        debug!("`TestHeaderProp` setter called, value: {}, header: {:?}, connection: {:?}, object_server: {:?}, emitter: {:?}",
+            value, header, connection, object_server, emitter
+        );
```

Comment:
best to do a `header.is_some();` here too.

Reply:
it would output on the logging but it wouldn't ensure that it's correctly set to `Some`.

Reply:
ah, there was a misunderstanding. Seems, also in case of the property getter test you added in the previous PR. I meant, we should check for it: `assert!(header.is_some());`.

Reply:
Right, that makes sense! Done.